cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/upgrade/DataMigration26.java

```diff
@@ -0,0 +1,282 @@
+                                                                            DatasetProperties.EMPTY)));
+      scopeToAggregatesTable.put(scope, getOrCreateMetricsTable(scope + AGGREGATES_TABLE_NAME_SUFFIX,
+                                                                DatasetProperties.EMPTY));
+      scopeToCodec.put(scope, new MetricsEntityCodec(scopeToEntityTable.get(scope),
+                                                     UpgradeMetricsConstants.DEFAULT_CONTEXT_DEPTH,
+                                                     UpgradeMetricsConstants.DEFAULT_METRIC_DEPTH,
+                                                     UpgradeMetricsConstants.DEFAULT_TAG_DEPTH));
+    }
+    aggMetricStore = new DefaultMetricStore(factory, new int[]{Integer.MAX_VALUE});
+  }
+
+  public void decodeAggregatesTable26() {
+    try {
+      for (String scope : scopes) {
+        Scanner scanner = scopeToAggregatesTable.get(scope).scan(null, null, null, null);
+        MetricsEntityCodec codec = scopeToCodec.get(scope);
+        LOG.info("Decoding for scope " + scope + "on table " + scopeToAggregatesTable.get(scope).toString() +
+                   " using codec " + codec.toString());
+        Row row;
+        while ((row = scanner.next()) != null) {
+          byte[] rowKey = row.getRow();
+          int offset = 0;
+          String context = codec.decode26(MetricsEntityType.CONTEXT, rowKey, offset);
+          offset += codec.getEncodedSize26(MetricsEntityType.CONTEXT);
+          String metricName = codec.decode26(MetricsEntityType.METRIC, rowKey, offset);
+          offset += codec.getEncodedSize26(MetricsEntityType.METRIC);
+          String runId = codec.decode26(MetricsEntityType.RUN, rowKey, offset);
+          constructMetricValue(scope, context, metricName, runId, row.getColumns().entrySet().iterator());
+        }
+      }
+    } catch (Exception e) {
+      LOG.warn("Exception during data-transfer in aggregates table", e);
+      // no-op
+    }
+  }
+
+  private void emitMetrics(Iterator<Map.Entry<byte[], byte[]>> iterator, String context, String metricName,
+                           Map<String, String> tagMap, String nonEmptyTagKey) throws Exception {
+    while (iterator.hasNext()) {
+      Map.Entry<byte[], byte[]> entry = iterator.next();
+      String tagValue = Bytes.toString(entry.getKey());
+      if (metricName.equals("dataset.store.bytes")) {
+        LOG.info("Dataset context : {} datasetName : {} value : {} run-id : {}", context, tagValue,
+                 Bytes.toLong(entry.getValue()), tagMap.get(Constants.Metrics.Tag.RUN_ID));
+      }
+      if (nonEmptyTagKey != null) {
+        if (tagValue.equals(UpgradeMetricsConstants.EMPTY_TAG)) {
+          continue;
+        } else {
+          long value = Bytes.toLong(entry.getValue());
+          Map<String, String> newMap = Maps.newHashMap(tagMap);
+          newMap.put(Constants.Metrics.Tag.NAMESPACE, Constants.DEFAULT_NAMESPACE);
```

Comment:
the nonEmptyTagKey, which i have renamed to metricTagType now,  if its non-null, we know we have emitted with a tag [ queue-name,dataset-name, stream-name] all of these belong to default namespace , right ? let me know if i am missing anthing